### ISSUE TITLE
oEmbed failures related to thumbnails

### ISSUE THREAD

**Issue by pySilver** [wagtail/wagtail#6788]
### Issue Summary

Trying to embed 2 items to test fixes recently introduced to Wagtail 2.12 (thumbnail hash):

https://twitter.com/CyberpunkGame/status/1354769450726285312
https://www.facebook.com/martyna.world/posts/256722362490410

Both fails with the following error:

```
Traceback (most recent call last):
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/query.py", line 573, in get_or_create
django_1          |     return self.get(**kwargs), False
django_1          |   File "/opt/venv/lib/python3.8/site-packages/cacheops/query.py", line 353, in get
django_1          |     return qs._no_monkey.get(qs, *args, **kwargs)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/query.py", line 429, in get
django_1          |     raise self.model.DoesNotExist(
django_1          | wagtail.embeds.models.Embed.DoesNotExist: Embed matching query does not exist.
django_1          | 
django_1          | During handling of the above exception, another exception occurred:
django_1          | 
django_1          | Traceback (most recent call last):
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
django_1          |     return self.cursor.execute(sql, params)
django_1          | psycopg2.errors.NotNullViolation: null value in column "thumbnail_url" violates not-null constraint
django_1          | DETAIL:  Failing row contains (23, https://twitter.com/CyberpunkGame/status/1354769450726285312, null, rich, <blockquote class="twitter-tweet"><p lang="en" dir="ltr">Hotfix ..., , Cyberpunk 2077, Twitter, null, 550, null, 2021-02-04 20:10:53.143013+00, bbcc5b725a47235ef04e65ddf094e6fb).
django_1          | 
django_1          | 
django_1          | The above exception was the direct cause of the following exception:
django_1          | 
django_1          | Traceback (most recent call last):
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
django_1          |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
django_1          |   File "/usr/local/lib/python3.8/contextlib.py", line 75, in inner
django_1          |     return func(*args, **kwds)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
django_1          |     response = view_func(request, *args, **kwargs)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/wagtail/admin/urls/__init__.py", line 127, in wrapper
django_1          |     return view_func(request, *args, **kwargs)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/wagtail/admin/auth.py", line 172, in decorated_view
django_1          |     response = view_func(request, *args, **kwargs)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/wagtail/embeds/views/chooser.py", line 29, in chooser_upload
django_1          |     embed_html = embed_to_editor_html(form.cleaned_data['url'])
django_1          |   File "/opt/venv/lib/python3.8/site-packages/wagtail/embeds/format.py", line 21, in embed_to_editor_html
django_1          |     embed = embeds.get_embed(url)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/wagtail/embeds/embeds.py", line 46, in get_embed
django_1          |     embed, created = Embed.objects.get_or_create(
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
django_1          |     return getattr(self.get_queryset(), name)(*args, **kwargs)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/query.py", line 576, in get_or_create
django_1          |     return self._create_object_from_params(kwargs, params)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/query.py", line 610, in _create_object_from_params
django_1          |     obj = self.create(**params)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/query.py", line 447, in create
django_1          |     obj.save(force_insert=True, using=self.db)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/base.py", line 753, in save
django_1          |     self.save_base(using=using, force_insert=force_insert,
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/base.py", line 790, in save_base
django_1          |     updated = self._save_table(
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/base.py", line 895, in _save_table
django_1          |     results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/base.py", line 933, in _do_insert
django_1          |     return manager._insert(
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
django_1          |     return getattr(self.get_queryset(), name)(*args, **kwargs)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/query.py", line 1254, in _insert
django_1          |     return query.get_compiler(using=using).execute_sql(returning_fields)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/models/sql/compiler.py", line 1397, in execute_sql
django_1          |     cursor.execute(sql, params)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 98, in execute
django_1          |     return super().execute(sql, params)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/cacheops/transaction.py", line 93, in execute
django_1          |     result = self._no_monkey.execute(self, sql, params)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 66, in execute
django_1          |     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
django_1          |     return executor(sql, params, many, context)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
django_1          |     return self.cursor.execute(sql, params)
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
django_1          |     raise dj_exc_value.with_traceback(traceback) from exc_value
django_1          |   File "/opt/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
django_1          |     return self.cursor.execute(sql, params)
django_1          | django.db.utils.IntegrityError: null value in column "thumbnail_url" violates not-null constraint
django_1          | DETAIL:  Failing row contains (23, https://twitter.com/CyberpunkGame/status/1354769450726285312, null, rich, <blockquote class="twitter-tweet"><p lang="en" dir="ltr">Hotfix ..., , Cyberpunk 2077, Twitter, null, 550, null, 2021-02-04 20:10:53.143013+00, bbcc5b725a47235ef04e65ddf094e6fb).
django_1          | 
django_1          | Internal Server Error: /cms/embeds/chooser/upload/ exc_info=(<class 'django.db.utils.IntegrityError'>, IntegrityError('null value in column "thumbnail_url" violates not-null constraint\nDETAIL:  Failing row contains (23, https://twitter.com/CyberpunkGame/status/1354769450726285312, null, rich, <blockquote class="twitter-tweet"><p lang="en" dir="ltr">Hotfix ..., , Cyberpunk 2077, Twitter, null, 550, null, 2021-02-04 20:10:53.143013+00, bbcc5b725a47235ef04e65ddf094e6fb).\n'), <traceback object at 0x7f1828201e00>)
django_1          | "POST /cms/embeds/chooser/upload/ HTTP/1.1" 500 276446
```

### Technical details

* Python version: 3.8.x
* Django version: 3.1.6
* Wagtail version: 2.12

**Comment by pySilver** [comment]
This are the output from embed parsers:

```
In [7]: finder("https://www.facebook.com/martyna.world/posts/256722362490410")
Out[7]:
{'title': '',
 'author_name': 'Martyna Wojciechowska',
 'provider_name': 'Facebook',
 'type': 'rich',
 'thumbnail_url': None,
 'width': 552,
 'height': None,
 'html': '<div id="fb-root"></div>\n<script async="1" defer="1" crossorigin="anonymous" src="https://connect.facebook.net/pl_PL/sdk.js#xfbml=1&amp;version=v9.0" nonce="mHQH7jyb"></script><div class="fb-post" data-href="https://www.facebook.com/martyna.world/posts/256722362490410" data-width="552"><blockquote cite="https://graph.facebook.com/100044579432860/posts/256722362490410/" class="fb-xfbml-parse-ignore"><p>Smutne wieści z Meksyku... \nZmarła MARBELLA AGUILAR, jedna z bohaterek programu KOBIETA NA KRAŃCU ŚWIATA, której...</p>Posted by <a href="https://www.facebook.com/martyna.world">Martyna Wojciechowska</a> on&nbsp;<a href="https://graph.facebook.com/100044579432860/posts/256722362490410/">Thursday, February 4, 2021</a></blockquote></div>'}

In [8]: finder("https://twitter.com/CyberpunkGame/status/1354769450726285312")
Out[8]:
{'title': '',
 'author_name': 'Cyberpunk 2077',
 'provider_name': 'Twitter',
 'type': 'rich',
 'thumbnail_url': None,
 'width': 550,
 'height': None,
 'html': '<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Hotfix 1.11 is available on PC, consoles and Stadia!<br><br>This update restores item randomization and fixes a bug which affected some users’ holocall with Takemura in Down on the Street quest.<br><br>Details: <a href="https://t.co/SsVYRGfdha">https://t.co/SsVYRGfdha</a> <a href="https://t.co/TAQmRilNxq">pic.twitter.com/TAQmRilNxq</a></p>&mdash; Cyberpunk 2077 (@CyberpunkGame) <a href="https://twitter.com/CyberpunkGame/status/1354769450726285312?ref_src=twsrc%5Etfw">January 28, 2021</a></blockquote>\n<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>\n'}
```

**Comment by allcaps** [comment]
We switched form `null=True, blank=True` to `blank=True` only for this field. Normally sting based fields should not be nullable. I assumed `thumbnail_url` was only nullable for unique together purposes. Not realising that embed provider would return `None`.

Two solutions:
1. We can revert the change and make `thumbnail_url` `null=True`. We should test what None means for the hash.
2. Cast None to an empty string before saving it.

I'll discuss with @gasman

**Comment by gasman** [comment]
Fixed in #6791
